src/chainstate/stacks/db/transactions.rs

```diff
@@ -1122,8 +1123,10 @@ impl StacksChainState {
         }
     }
 
-    /// Process a transaction.  Return the fee and the transaction receipt
-    pub fn process_transaction(
+    /// This method implements the contract of `process_transaction`, and
+    /// exists as an intermediate step so that its output can be captured and
+    /// logged.
+    fn process_transaction_internal(
```

Comment:
The naming convention in this codebase for such "internal" functions is to prefix it with `inner_`.  This should be `inner_process_transaction`.